### PR TITLE
Simplify address parsing

### PR DESCRIPTION
**Pull request by cdecker**
This PR picks up some work by @bisoge in PR #1850 which I found useful. It now has a clearer separation between address parsing and deciding whether an address matches our current chain. I took the liberty of cherry-picking a bit and inverting the way we check for chain compatibility, but the inspiration came from PR #1850.

Closes #1850

### CODE REVIEW

File: bitcoin/base58.c [hunk 1]
```diff
@@ -67,40 +67,24 @@ static bool from_base58(u8 *version,
 	return true;
 }
 
-bool bitcoin_from_base58(bool *test_net,
-			 struct bitcoin_address *addr,
+bool bitcoin_from_base58(u8 *version, struct bitcoin_address *addr,
```
Comment (niftynei) [inline bitcoin/base58.c]: is it worth preserving this method (and the ones below) if they're just a wrapper for another?

**Comment by niftynei** [comment]
you've got a build error tho: `external/libbase58 is not a submodule!`

**Comment by rustyrussell** [comment]
> you've got a build error tho: `external/libbase58 is not a submodule!`

Some weird Travis caching thing.  Blowing away the cache seems to have fixed (hit this on another PR)

File: bitcoin/base58.c (hunk 1 above)
Comment (cdecker) [reply bitcoin/base58.c]: I was tempted to remove them, however the type-safety that this method gives (`struct bitcoin_address` wraps `struct ripemd160`) is worth it I think.

**Pull request merged**